From: Mike de Boer <mdeboer@mozilla.com>
Date: Wed, 22 May 2019 09:46:58 +0000
Subject: Bug 1553413 - Update and harden form data filtering for privacy to
 account for no data being passed in. r=dao

Differential Revision: https://phabricator.services.mozilla.com/D32116
---
 toolkit/modules/sessionstore/PrivacyFilter.jsm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/modules/sessionstore/PrivacyFilter.jsm b/toolkit/modules/sessionstore/PrivacyFilter.jsm
index 88a94d11fe65..c7b4b86c49c9 100644
--- a/toolkit/modules/sessionstore/PrivacyFilter.jsm
+++ b/toolkit/modules/sessionstore/PrivacyFilter.jsm
@@ -46,7 +46,7 @@ var PrivacyFilter = Object.freeze({
     // If the given form data object has an associated URL that we are not
     // allowed to store data for, bail out. We explicitly discard data for any
     // children as well even if storing data for those frames would be allowed.
-    if (data.url && !PrivacyLevel.check(data.url)) {
+    if (!data || (data.url && !PrivacyLevel.check(data.url))) {
       return null;
     }
 
